### PR TITLE
Decaffeinate 2020 is operator with overloading

### PR DESCRIPTION
**Pull request by okeefm**
Final PR to merge the `Is`, `As`, and `function overloading` work down to `decaffeinate-2020`.

Pull requests into cql-execution require the following.
Submitter and reviewer should ✔ when done.
For items that are not-applicable, mark "N/A" and ✔.

[CDS Connect](https://cds.ahrq.gov/cdsconnect) and [Bonnie](https://github.com/projecttacoma/bonnie) are the main users of this repository. 
It is strongly recommended to include a person from each of those projects as a reviewer.

**Submitter:**
- [ ] This pull request describes why these changes were made
- [ ] Code diff has been done and been reviewed (it does not contain: additional white space, not applicable code changes, debug statements, etc.)
- [ ] Tests are included and test edge cases
- [ ] Tests have been run locally and pass
- [ ] Code coverage has not gone down and all code touched or added is covered.
- [ ] All dependent libraries are appropriately updated or have a corresponding PR related to this change
- [ ] `cql4browsers.js` built with `yarn run build-everything` if coffeescript source changed.

**Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: src/elm/reusable.js [hunk 1]
```diff
@@ -50,20 +50,43 @@ class FunctionRef extends Expression {
...
+
+    // Filter out functions w/ wrong number of arguments.
+    functionDefs = functionDefs.filter(f => f.parameters.length === args.length);
+    // If there is still > 1 matching function, filter by argument types
+    if (functionDefs.length > 1) {
+      functionDefs = functionDefs.filter(f => {
+        let match = true;
+        for (let i = 0; i < args.length && match; i++) {
+          if (args[i] !== null) {
+            match = ctx.matchesTypeSpecifier(args[i], f.parameters[i].operandTypeSpecifier);
```
Comment (cmoesel) [inline src/elm/reusable.js:73]: NOTE: I don't know if it is necessary, but if we wanted to be extra safe, we could check if `operandTypeSpecifier` is null, and if so, convert `operandType` to a `NamedTypeSpecifier` (similar to [here](https://github.com/cqframework/cql-execution/blob/08bf598ed5093a463eedc90ea6c3baa0fd829b58/src/elm/type.js#L16-L22)) (assuming `operandType` is not also null).

But I _think_ in practice CQL to ELM only ever uses `operandTypeSpecifier`.

**Review by cmoesel — APPROVED** [review]
This is a great capability to have in cql-execution.  Thank you @okeefm and @hossenlopp!

I included one note about a potential improvement.  I can make it, if you'd like.  Or we can leave it as a task for later.

**Review by cmoesel — APPROVED** [review]
